src/kafka-util/src/client.rs

```diff
@@ -176,6 +176,9 @@ impl FromStr for MzKafkaError {
             Ok(Self::AllBrokersDown)
         } else if s.contains("Unknown topic or partition") || s.contains("Unknown partition") {
             Ok(Self::UnknownTopicOrPartition)
+        } else if s.contains("failed to connect to the remote host") {
+            // This is an error logged synthesized by `DynamicBrokerRewritingClientContext`
```

Comment:
Funky grammar. Also, can we say more here about why this is important? What goes wrong if we don't cast this error message to `Self::Internal`?

Reply:
this code is a bit weird; we do end up turning anything else into an `Internal`, but we print a warning; we know this error can happen, so it should have a variant

Reply:
I could have SWORN i removed this, yeah this makes no sense anymore